pkg/retry/retry.go

```diff
@@ -32,9 +32,15 @@ const (
 	DefaultMaxElapsedTime      = 1 * time.Minute
 )
 
-// Retry retries the given function DefaultRetryCount times while sleeping some time between unsuccessful attempts
+// Do retries the given function until DefaultMaxInterval time passes, while sleeping some time between unsuccessful attempts
 // if retryOnError returns true, continue retry, otherwise, return error
 func Do(operation func() error, retryOnError func(err error) bool) error {
+	return RetryWithConfiguredTime(operation, retryOnError, DefaultMaxElapsedTime)
+}
+
+// RetryWithConfiguredTime will retry with an exponential back off until "maxRetryTime" worth of sleeping has occurred.
+// this allows users of this function to customize how long they intend to retry a function before giving up on the operation.
+func RetryWithConfiguredTime(operation func() error, retryOnError func(err error) bool, maxRetryTime time.Duration) error {
```

Comment:
Alternatively we could introduce retry.Config with default values and change that when we want to use different values

```
type Config struct {
        Interval ...
        RandomizationFactor 
        etc....
}

DefaultConfig {
       Interval: DefaultInitialInterval  , 
       RandomizationFactor: 	DefaultRandomizationFactor
       etc..
}
```

We introduce DoWithConfig (or something like that) which accepts the config struct as a parameter and Do calls it with DefaultConfig